imap/imapd.c

```diff
@@ -15807,3 +15830,38 @@ static void cmd_getjmapaccess(char *tag)
     prot_printf(imapd_out, "%s OK %s\r\n", tag,
                 "This server is also accessible via JMAP, see RFC8620");
 }
+
+static void cmd_uidbatches(char *tag, uint32_t size, uint32_t low, uint32_t high)
+{
+    if (backend_current) {
+        /* remote mailbox */
+        prot_printf(backend_current->out, "%s UIDBATCHES %u", tag, size);
+        if (low > 1 || high < UINT32_MAX) {
+            prot_printf(backend_current->out, " %u:%u", low, high);
+        }
+        prot_puts(backend_current->out, "\r\n");
+        pipe_including_tag(backend_current, tag, 0);
+        return;
+    }
+
+    prot_printf(imapd_out, "* UIDBATCHES (TAG \"%s\") UID", tag);
+    if (low <= (imapd_index->exists + size - 1) / size) {
+        int64_t msgno = imapd_index->exists - ((low - 1) * size);
+        uint32_t batch = low;
+        char sep = ' ';
+
+        prot_puts(imapd_out, " ALL");
+
+        for (; msgno > 0 && batch <= high; batch++) {
```

Comment:
I think this loop body gets skipped if `low > high`, but we've already started an output line, does that result in us producing invalid output?